packages/angular/projects/clr-angular/src/data/datagrid/datagrid-row.ts

```diff
@@ -248,6 +247,15 @@ export class ClrDatagridRow<T = any> implements AfterContentInit, AfterViewInit
     this.subscriptions.forEach((sub: Subscription) => sub.unsubscribe());
   }
 
+  ngOnChanges(changes: any) {
+    /**
+     * When `clrDgSelectable` make sure  to lock or unlock the given row
+     */
+    if (changes.clrDgSelectable && changes.clrDgSelectable.currentValue !== changes.clrDgSelectable.previousValue) {
```

Comment:
Ok this is going to run _every change detection cycle_ for _every row_. Can we avoid this with some other approach?

Reply:
Are we talking about the `if`  or the body of the `if`? 

But you are right it's called for every row, at least once. `ngOnChanges` is called after all `inputs` are updated so the good thing is that won't run for every input update.

It's not that different than having it inside the `@Input` getter. 

I could think of something else as a way to monitor the change of  `clrDgSelectable` but right now I don't have any idea :) 

Open for suggestions.